typescript/infra/config/environments/mainnet3/gasPrices.json

```diff
@@ -1,20 +1,74 @@
 {
-  "ancient8": "1",
-  "arbitrum": "0.1",
-  "avalanche": "43.212830197",
-  "bsc": "1.350070319",
-  "celo": "10.0",
-  "ethereum": "26.346912847",
-  "mantapacific": "0.100000057",
-  "moonbeam": "125.0",
-  "optimism": "0.003225814",
-  "polygon": "61.601287856",
-  "gnosis": "1.852997796",
-  "base": "0.0010003",
-  "scroll": "0.46",
-  "polygonzkevm": "3.95",
-  "inevm": "0.1",
-  "viction": "0.25",
-  "neutron": "0.1",
-  "injective": "0.1"
+  "arbitrum": {
+    "amount": "0.01",
+    "decimals": 9
+  },
+  "ancient8": {
+    "amount": "0.001000252",
+    "decimals": 9
+  },
+  "avalanche": {
+    "amount": "43.212830197",
+    "decimals": 9
+  },
+  "bsc": {
+    "amount": "5.0",
+    "decimals": 9
+  },
+  "celo": {
+    "amount": "10.0",
+    "decimals": 9
+  },
+  "ethereum": {
+    "amount": "26.346912847",
+    "decimals": 9
+  },
+  "mantapacific": {
+    "amount": "0.100163166",
+    "decimals": 9
+  },
+  "moonbeam": {
+    "amount": "125.0",
+    "decimals": 9
+  },
+  "optimism": {
+    "amount": "0.061126811",
+    "decimals": 9
+  },
+  "polygon": {
+    "amount": "101.76455238",
+    "decimals": 9
+  },
+  "gnosis": {
+    "amount": "3.236596353",
+    "decimals": 9
+  },
+  "base": {
+    "amount": "0.116871685",
+    "decimals": 9
+  },
+  "scroll": {
+    "amount": "1.3231",
+    "decimals": 9
+  },
+  "polygonzkevm": {
+    "amount": "3.95",
+    "decimals": 9
+  },
+  "inevm": {
+    "amount": "0.1",
+    "decimals": 9
+  },
+  "viction": {
+    "amount": "0.25",
+    "decimals": 9
+  },
+  "neutron": {
+    "amount": "0.0053",
+    "decimals": 1
+  },
+  "injective": {
+    "amount": "700000000",
+    "decimals": 1
+  }
```

Comment:
yep you're exactly right, it just pads zeroes. For neutron, see the PR description:
> Handles decimals when it comes to gas prices. The existing tooling assumed gas prices would always be denominated in wei. An extra weird situation arises with non-EVM chains: chains like Neutron have gas prices that are quotes in the smallest denomination (ie the equivalent of "wei" for NTRN), but the gas price is not an integer and is very small, e.g. 0.0053. This poses issues because the IGP contracts require the gas price that's configured onchain to be an integer. Simply rounding up from 0.0053 to 1 results in almost 2 orders of magnitude difference :(. So the solution here is to have some logic that accommodates really tiny gas prices by scaling up the configured gas price and scaling down the configured token exchange rate.

or the comment here https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3662/files#diff-ac1f1ed84dad24fd8eeff4f7b636db47ba358af2aae94141c70bc757e58a59a3R27

It's super weird but somehow it's possible for the gas price to be < 1 and still be denominated in the smallest unit of NTRN - cosmos seemingly allows floating point gas prices

Reply:
ahhhh gotcha, I did read that in the description but didn't quite piece it together til now - thanks!